$OpenBSD: patch-src_libs_ec_cpp_ECTag_cpp,v 1.2 2012/06/05 08:42:32 dcoppa Exp $

Fix endianness bug in EC
(upstream git commit ec79c81d6825b55ed826b6519287112056859735)

--- src/libs/ec/cpp/ECTag.cpp.orig	Mon Jun 13 10:50:25 2011
+++ src/libs/ec/cpp/ECTag.cpp	Tue Jun  5 09:16:23 2012
@@ -227,13 +227,13 @@ void CECTag::InitInt(uint64 data)
 			PokeUInt8( m_tagData, (uint8) data );
 			break;
 		case EC_TAGTYPE_UINT16:
-			PokeUInt16( m_tagData, wxUINT16_SWAP_ALWAYS((uint16) data ));
+			RawPokeUInt16( m_tagData, ENDIAN_HTONS((uint16) data ));
 			break;
 		case EC_TAGTYPE_UINT32:
-			PokeUInt32( m_tagData, wxUINT32_SWAP_ALWAYS((uint32) data ));
+			RawPokeUInt32( m_tagData, ENDIAN_HTONL((uint32) data ));
 			break;
 		case EC_TAGTYPE_UINT64:
-			PokeUInt64( m_tagData, wxUINT64_SWAP_ALWAYS(data) );
+			RawPokeUInt64( m_tagData, ENDIAN_HTONLL(data) );
 			break;
 	}
 }
